Debug flag support for different Intel compiler versions

ESMA_cmake debug flags used with Intel compilers include unitialized variables check `-check uninit`. 

https://github.com/GEOS-ESM/ESMA_cmake/blob/c2e6fc5c9903624a443f96b3a50c62f22aad5a3d/Intel.cmake#L58

This works with ifort19 but causes an error with ifort18 when building MAPL, as reported by a GCHPctm user (https://github.com/geoschem/GCHPctm/issues/35). The error message is:

[ 15%] Building Fortran object src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o
ifort: error #10236: File not found:  'uninit'
make[2]: *** [src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o] Error 1
make[2]: *** Deleting file `src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/ESMF_CFIOBaseMod.f.o'
make[1]: *** [src/MAPL/MAPL_cfio_r4/CMakeFiles/MAPL_cfio_r4.dir/all] Error 2
make: *** [all] Error 2

Changing the flag `-check,uninit` fixes the problem, but then makes it incompatible with ifort19.

Do you have plans to allow for version-specific flags in ESMA_cmake to fix issues like this?

--- a/Intel.cmake
+++ b/Intel.cmake
@@ -55,7 +55,7 @@ set (common_Fortran_fpe_flags "${FPE0} ${FP_MODEL_SOURCE} ${HEAPARRAYS} ${NOOLD_
 
 # GEOS Debug
 # ----------
-set (GEOS_Fortran_Debug_Flags "${DEBINFO} ${FOPT0} ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -debug -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -check bounds -check uninit -fp-stack-check -warn unused -init=snan,arrays -save-temps")
+set (GEOS_Fortran_Debug_Flags "${DEBINFO} ${FOPT0} ${FTZ} ${ALIGN_ALL} ${NO_ALIAS} -debug -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -check all,noarg_temp_created -fp-stack-check -warn unused -init=snan,arrays -save-temps")
 set (GEOS_Fortran_Debug_FPE_Flags "${common_Fortran_fpe_flags}")
 
 # GEOS Release

